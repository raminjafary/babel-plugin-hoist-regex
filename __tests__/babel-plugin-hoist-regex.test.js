const babel = require('@babel/core')
const babelPluginHoistRegex = require('../lib')

const example = `
function ramin(numberLiteral) {
  const regex = /[0-9]+/
  const {input} = regex.exec(numberLiteral)
  const a = () => {
    const rgx = /[0-9]+/
    const {input} = regex.exec(numberLiteral)
  }
  return input
}
`

describe('babelPluginHoistRegex Test', () => {
  test('Snapshot', () => {
    const {code} = babel.transform(example, {plugins: [babelPluginHoistRegex]})
    expect(code).toMatchSnapshot()
  })

  test('AST', () => {
    const {ast} = babel.transform(example, {
      plugins: [babelPluginHoistRegex],
      ast: true
    })
    const program = ast.program
    const declaration = program.body[0].declarations[0]
    expect(declaration.id.name).toEqual('_rgx')
    // or babelTraverse(program, {visitor: ...})
  })
  test('Exec', () => {
    const {code} = babel.transform(example, {plugins: [babelPluginHoistRegex]})
    const f = new Function(`
    ${code};
    return regex = _regex;
  `)
    const regex = f()
    expect(regex).toBe(regex)
  })
})
