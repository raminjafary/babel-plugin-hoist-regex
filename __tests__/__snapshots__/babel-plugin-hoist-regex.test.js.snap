// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babelPluginHoistRegex Test Snapshot 1`] = `
"const _rgx = /[0-9]+/;
const _regex = /[0-9]+/;

function ramin(numberLiteral) {
  const {
    input
  } = _regex.exec(numberLiteral);

  const a = () => {
    const {
      input
    } = _regex.exec(numberLiteral);
  };

  return input;
}"
`;
